pkg/controller/syncsetinstance/syncsetinstance_controller.go

```diff
@@ -656,6 +650,7 @@ func (r *ReconcileSyncSetInstance) applySyncSetResources(ssi *hivev1.SyncSetInst
 		}
 	}
 
+	ssi.Status.Conditions = r.clearUnknownObjectSyncCondition(ssi.Status.Conditions)
```

Comment:
There can be a situation wherein we are able to get information about the resource but are not able to apply it. In this case UnknownObjectCondition won't be cleared immediately. I do not know how to handle that yet.

Reply:
I think that the UnknownObjectCondition should be moved from a condition on the entire SyncSetInstance to a condition on each of the individual resources in the SyncSetInstance. In other words, the condition should be set on `.status.resources[i].conditions` rather than on `.status.conditions`.